extension-adinfo/src/adinfo.cpp

```diff
@@ -6,27 +6,42 @@
 
 #define DLIB_LOG_DOMAIN LIB_NAME
 #include <dmsdk/sdk.h>
-#include "private_adinfo.h"
+#include "adinfo.h"
+
+
+
+
+ADInfoData g_ADInfoData;
+
+void ADInfo_QueueAdId(const char* adId) {
+    g_ADInfoData.m_AdId = adId;
```

Comment:
Is this string guaranteed to live for this duration of time between this call and the callback?

Reply:
You're right. Ownership is unclear. I'll make a copy!